### PR TITLE
Add documentation for `platform_view` example.

### LINKED ISSUE #111142: Add documentation to macOS `platform_view` example.
## Use case

Developers examining the `platform_view` example do not have documentation that explains platform-specific code more clearly.

Making this a follow up of nit-comment by @cbracken  https://github.com/flutter/flutter/pull/111005#pullrequestreview-1099701873.

## Proposal

Add more documentation to `examples/platform_view/macos/*`

### PR DESCRIPTION
**Pull request by a-wallen**
When `platform_view` first landed for `macOS` there was no documentation for any of the classes added. This PR adds documentation for the `MainFlutterWindow` and the `PlatformViewController` classes.

fixes https://github.com/flutter/flutter/issues/111142

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
> [Tree Hygiene page](https://github.com/flutter/flutter/wiki/Tree-hygiene#how-to-review-code): only affect comments (including documentation).

- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by a-wallen** [comment]
Dear auto-submit bot,

Please read my PR and see that it only affects comments.

Thanks!
Alex

File: examples/platform_view/macos/Runner/MainFlutterWindow.swift [hunk 1]
```diff
@@ -5,6 +5,11 @@
 import Cocoa
 import FlutterMacOS
 
+/**
+The window that is automatically generated when `flutter create --target=macos`
+on a project. `MainFlutterWindow` uses a FlutterViewController as it's content
+view controller by default.
+*/
```
Comment (cbracken) [inline examples/platform_view/macos/Runner/MainFlutterWindow.swift:12]: Maybe something like:

```swift
/**
 The main application window.

 Performs Flutter app initialization, and handles channel method calls over the
 `samples.flutter.io/platform_view` channel.
*/
```

File: examples/platform_view/macos/Runner/PlatformViewController.swift [hunk 2]
```diff
@@ -4,6 +4,18 @@
...
+the flutter view for this application. ViewControllers that inherit from this
+class should be displayed as a popover or modal, with a button that binds to
+the IBAction `pop()`.
+
+To get the value of the popover during close, pass a callback function as
+the `dispose` parameter. The callback passed will have access to the
+`PlatformViewController` and all of it's properties at close so that the `count`
+can be passed back through the message channel if needed.
+*/
+
```
Comment (cbracken) [inline examples/platform_view/macos/Runner/PlatformViewController.swift:18]:
```suggestion
```

**Review by cbracken — APPROVED** [review]
LGTM modulo comments!

**Pull request merged**